tests/test_load.py

```diff
@@ -245,6 +246,42 @@ def test_load_dataset_local_with_default_in_memory(
     assert (dataset["train"].dataset_size < max_in_memory_dataset_size) is expected_in_memory
 
 
+class TestLoadDatasetOnlySplits:
+    def test_load_dataset_local_only_splits_processed_files(self, dataset_loading_script_dir, data_dir, tmp_path):
+        download_config = DownloadConfig(splits=["test"])
+        cache_dir = str(tmp_path / "cache")
+        datasetdict = datasets.load_dataset(
+            dataset_loading_script_dir,
+            data_dir=data_dir,
+            cache_dir=cache_dir,
+            download_config=download_config,
+        )
```

Comment:
In my opinion the users shouldn't have another place to choose the split. The `split` parameter of `load_dataset` is made for this.

Reply:
As far as I know, the `split` parameter of `load_dataset` is for _loading_: loading from cache.

On the contrary, the `splits` parameter passed to `DownloadConfig` is for _downloading_.